### PR TITLE
DateTime class wrapper

### PR DESCRIPTION
**Pull request by eucalvo**
Added the wrapper and tests for DateTime class.

### CODE REVIEW

File: test/DateTime.cpp [hunk 1]
```diff
@@ -28,4 +48,17 @@ unittest(yearMonthAsPath) {
   assertEqual("/2020/09", buffer);
 }
 
+unittest(setAsCurrent) {
+  DateTime_TC jan(2021, 1, 15, 1, 48, 24);
+  DateTime_TC dob(2001, 2, 20, 3, 50, 10);
```
Comment (jgfoster) [inline test/DateTime.cpp]: To minimize revealing info, let's just use `jan` as "now" and adjust following lines of asserts.

File: test/DateTime.cpp [hunk 2]
```diff
@@ -3,18 +3,38 @@
...
   DateTime_TC now = DateTime_TC::now();
   assertTrue(now.year() >= 2020);
+  assertTrue(now.month() >= 1);
+  assertTrue(now.day() >= 1);
+  assertTrue(now.hour() >= 0);
+  assertTrue(now.minute() >= 0);
+  assertTrue(now.second() >= 0);
 }
 
 //  void printToSerial();   // DigitalClockDisplay.ino
```
Comment (jgfoster) [inline test/DateTime.cpp]: This code came from ... in the older version of TankController.ino.

File: test/DateTime.cpp [hunk 3]
```diff
@@ -3,18 +3,38 @@
...
 
-/*
-bundle exec arduino_ci.rb --skip-examples-compilation --testfile-select=DateTime.cpp
-*/
+/**
+ * year has to be range: 2000-2099
+ * following command only runs this test
+ * bundle exec arduino_ci.rb --skip-examples-compilation
+ * --testfile-select=DateTime.cpp
+ */
```
Comment (jgfoster) [inline test/DateTime.cpp:12]: singleton test that the function returns the identical object each time.

**Review by jgfoster — CHANGES_REQUESTED** [review]

**Review by prestoncarman — APPROVED** [review]
LGTM

**Review by jgfoster — APPROVED** [review]
LGTM

**Pull request merged**